src/mrack/providers/openstack.py

```diff
@@ -545,13 +546,38 @@ async def create_server(self, req):
             except ServerError as exc:
                 logger.debug(exc)
                 error_attempts += 1
-                if error_attempts > SERVER_ERROR_RETRY:
-                    raise ProvisioningError(
-                        f"{self.dsp_name}: Fail to create server", specs
-                    ) from exc
-                await asyncio.sleep(SERVER_ERROR_SLEEP)
+                if error_attempts <= SERVER_ERROR_RETRY:
+                    await asyncio.sleep(SERVER_ERROR_SLEEP)
+                    continue  # Try again due to ServerError
+
+            if error_attempts > SERVER_ERROR_RETRY:
+                # now we are past to what we would like to wait fail now
+                raise ProvisioningError(
```

Comment:
Not related to your changes, but this exception is not being handled, so mrack execution exits here, I don't remember if this is expected. Output with forced 500 error: [mrack.log](https://github.com/neoave/mrack/files/6467171/mrack.log)

Reply:
Oh so maybe we would like to handle it somehow...

Reply:
feat: gracefully destroy servers after traceback

This allows us to create dummy Host object based
on the ProvisioningError traceback arguments
where we added also requirement which may have
caused the provisioning to fail so the worflow
can continue and remove already provisioned
servers after reprovision / abort strategy is
done so resources that may not be freed
to openstack will be returned.